### PR TITLE
New Integration Tests

### PR DESCRIPTION
**Pull request by colin99d**
# Description

This allows dynamic argument naming in the integrated tests and also makes the tests much easier to work with. Before merge I need to finish the following things:

- [x] Create a more detailed CSV Output
- [x] Understand how we can run this on the installer @simmonsj330
- [x] Combine my new documentation with this existing documentation (@simmonsj330 and I still need to write the installer docs)
- [x] Implement a verbose option

Also, this PR now saves terminal output from the tests in the `integration_test_output` folder, for viewing after tests run.

# How has this been tested?

* Please describe the tests that you ran to verify your changes.
* Provide instructions so we can reproduce.
* Please also list any relevant details for your test configuration.
- [x] Make sure affected commands still run in terminal
- [x] Run the integration tests

# Checklist:

- [x] Update [our Hugo documentation](https://openbb-finance.github.io/OpenBBTerminal/) following [these guidelines](https://github.com/OpenBB-finance/OpenBBTerminal/tree/main/website).
- [x] Update our tests following [these guidelines](https://github.com/OpenBB-finance/OpenBBTerminal/tree/main/tests).
- [x] Make sure you are following our [CONTRIBUTING guidelines](https://github.com/OpenBB-finance/OpenBBTerminal/blob/main/CONTRIBUTING.md).
- [x] If a feature was added make sure to add it to the corresponding [scripts file](https://github.com/OpenBB-finance/OpenBBTerminal/tree/main/scripts).

# Others
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly in hard-to-understand areas.

### CODE REVIEW

**Comment by colin99d** [comment]
@simmonsj330 and I both are running into issues with integration tests "snagging", I am running them one section at a time to learn more.

**Comment by simmonsj330** [comment]
@colin99d What about stocks?

**Comment by colin99d** [comment]
> @colin99d What about stocks?

<iframe style="border-radius:12px" src="https://open.spotify.com/embed/track/3QHOeZLrjWV3R1ps587v7c?utm_source=generator" width="100%" height="352" frameBorder="0" allowfullscreen="" allow="autoplay; clipboard-write; encrypted-media; fullscreen; picture-in-picture" loading="lazy"></iframe>

**Comment by colin99d** [comment]
@simmonsj330, I eventually got eveything else running. Should we look into how we wanna run tests with installer and then get this thing merged?

**Comment by colin99d** [comment]
@simmonsj330 looking through, I am able to find functions that stick, but these seems to be an issue on the terminal, and not with this PR. I think we should merge.

**Comment by colin99d** [comment]
Screenshot 2022-11-25 at 10 16 37 AM
For example, here is the PO menu hanging. But this PR only helps diagnose the issue, and does nothing to cause this hang.

**Comment by colin99d** [comment]
Screenshot 2022-11-25 at 11 56 05 AM
Looks like tests also work fine in our build CI with the installer.

File: openbb_terminal/integration_testing.py [hunk 1]
```diff
@@ -0,0 +1,213 @@
...
+        formatter_class=argparse.ArgumentDefaultsHelpFormatter,
+        prog="testing",
+        description="Integration tests for the OpenBB Terminal.",
+    )
+    parser.add_argument(
+        "-f",
+        "--file",
+        help=(
+            "The path or .openbb file to run. Starts at "
+            "OpenBBTermina/openbb_terminal/miscellaneous/scripts"
```
Comment (jmaslek) [inline openbb_terminal/integration_testing.py:161]: typo

File: openbb_terminal/miscellaneous/scripts/README.md [hunk 2]
```diff
@@ -28,6 +28,10 @@ should be `test_<menu>_<command>.openbb` if you are testing a specific command o
 if you are testing the entire menu. However, it is encouraged to create as specific of integration tests
 as possible to identify errors more precisely. Additionally, all tests must end with the `exit` command.
 
+These files can be given dynamic output with the following syntax `${key=default}`. Please note that
+both key and default can only contain letters and numbers with NO special characters. Each dynamic
+argument MUST contain a key and a default value.
```
Comment (jmaslek) [inline openbb_terminal/miscellaneous/scripts/README.md:33]: Plz include an example

**Review by jmaslek — APPROVED** [review]

File: openbb_terminal/miscellaneous/scripts/README.md (hunk 2 above)
Comment (colin99d) [reply openbb_terminal/miscellaneous/scripts/README.md:33]: Are the examples below not enough?

**Pull request merged**